gtdynamics/utils/Trajectory.cpp

```diff
@@ -36,6 +36,21 @@ using std::vector;
 
 namespace gtdynamics {
 
+
+void Trajectory::addPhaseContactPoints(const Phase &phase) {
+  // Add unique PointOnLink objects to contact_points_
+  for (auto &&kv : boost::static_pointer_cast<const FootContactState>(phase.constraints())->contactPoints()) {
+    int link_count =
```

Comment:
did I see this code before? (i.e. copy/paste alert?)

Reply:
I don't think so.
There is some similarity with `FootContactConstraintSpec::HasContact` but its not the same.